### PR TITLE
Python patcher compatibility

### PR DESCRIPTION
**Pull request by 4z0t**
Includes changes for python patcher to work:
* relative includes
* old ways to register lua function
* proper type conversions
* removed LuaJIT DLL load

### CODE REVIEW

**Review by Hdt80bro — APPROVED** [review]

File: section/Init.cpp [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#include "moho.h"
+
+extern const uintptr_t FIRST_CTOR;
```
Comment (Garanas) [inline section/Init.cpp:3]: Is the behavior of this documented? As in, what does this new file do?

File: section/LuaExts.cpp [hunk 2]
```diff
@@ -1,22 +1,21 @@
...
-            //PatcherList_LCreateTable
-            extern void *LCreateTable[];
-            if (LCreateTable[0])
-                lua_createtable = LCreateTable[0];
-            GetTableAH = e_GetTableAH;
-        }
-    }
-} luaexts;
\ No newline at end of file
+// struct LuaExts {
```
Comment (Garanas) [inline section/LuaExts.cpp:8]: It is better to remove commented code. We have Git history to retrieve it.

**Review by Garanas — COMMENTED** [review]
It would be interesting to have some form of documentation of how certain procedures work that are used often. For example: registering a new sim or UI function.

**Comment by 4z0t** [comment]
> It would be interesting to have some form of documentation of how certain procedures work that are used often. For example: registering a new sim or UI function.
> 
> I of course can not read and understand the code exactly. But in general these changes are related to fixing paths and/or types. There appear to be just a few large convention changes, such as the `Init.cpp` file.

TBH I'd leave it for another PR (this one https://github.com/FAForever/FA-Binary-Patches/pull/102 for example)

File: section/Init.cpp (hunk 1 above)
Comment (4z0t) [reply section/Init.cpp]: This file calls global variables constructors.

**Pull request merged**